### PR TITLE
ASI Filter Wheel, Save GUI Settings on Exit, and Microscope Object

### PR DESCRIPTION
**Pull request by AdvancedImagingUTSW**
Addresses:
#447 - Added functionality to the controller that now overwrites the .config/experiment.yml file upon exit. Thus, when the program is reloaded, it is in the same state that it was exited in. 

#429 - ASI Filter Wheel Commands now added. Only one instance of the device_connection can exist for the ASI Tiger Controller, which controls both the filter wheels, dichroic changers (not yet implemented), and stages.  Cleaned up the Microscope object to make the dictionary parsing and logic slightly more clear (broke it up into a few more sub functions). I believe the logic now makes sense to me, but I will need to make sure that the ASI device_connection gets passed to both the stages and filter wheels in the morning. 

Also, given that I mucked around with the microscope object, which is obviously critical, we should test this on more than one microscope.

### CODE REVIEW

File: src/aslm/model/device_startup_functions.py [hunk 1]
```diff
@@ -287,19 +287,23 @@ def load_stages(configuration, is_synthetic=False):
...
-
-            stage_devices.append(
-                auto_redial(
-                    build_ASI_Stage_connection,
-                    (
-                        stage_config["port"],
-                        stage_config["baudrate"],
-                    ),
-                    exception=TigerException,
+            filter_wheel = configuration["configuration"]["hardware"]["filter_wheel"]['type']
```
Comment (annie-xd-wang) [inline src/aslm/model/device_startup_functions.py:290]: This is assuming that if there is an ASI filter wheel, the filter wheel will be the same hardware of an ASI stage. It works, but I think we'd better refactor current device loading structure to make sure devices connect to the right hardware without hard coding.

**Review by annie-xd-wang — APPROVED** [review]
I've read the code, but have the feeling that we should refactor the devices loading structure. Should we refactor before merging this branch? or create a new issue and refactor later?

File: src/aslm/model/device_startup_functions.py (hunk 1 above)
Comment (AdvancedImagingUTSW) [reply src/aslm/model/device_startup_functions.py:290]: I think it is quite difficult to account for every single permutation. Technically, there could be three different scenarios for an ASI filter wheel, and they would have to be handled differently.

ASI offers an MS2000 controller, a Tiger Controller, and an independent Filter Wheel controller. Some of these have different command sets. So in the meantime, we will have to stick with a Tiger Controller.

**Comment by AdvancedImagingUTSW** [comment]
I'm somewhat tempted to create an issue that lists various things that should be refactored. But then again, I understand that if it doesn't happen soon, it probably won't happen.

How long do you think it would take to refactor it?

**Pull request merged**